release/run_release_test.sh

```diff
@@ -50,9 +50,7 @@ if [ -n "${RAY_COMMIT_OF_WHEEL-}" ]; then
 fi
 
 if [ -z "${NO_INSTALL}" ]; then
-  pip install --use-deprecated=legacy-resolver -q -r requirements.txt
```

Comment:
Do we still need `requirements.txt` after removing it here?

Reply:
I am hoping the answer is no.

all our tests are byod now. this requirement file is only used outside the job's docker.

Reply:
we will run some tests and see.

probably will not cherrypick this if tests are already passing now.

Reply:
My guess is that it's here because folks some time still run tests via ray job instead of anyscale job; but I think we can deprecate ray job as a way of running release tests at this point and we can simplify a lot the infrastructure

@krfricke do you think it should be fine to have anyscale as the only way to run release tests at this point, thankks